google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageRetryHandler.java

```diff
@@ -34,26 +30,45 @@
   private long totalWaitTime; // in milliseconds
   private final int maxRetries;
   private final int maxReopens;
+  private final CloudStorageConfiguration config;
+
+  /**
+   * Create a CloudStorageRetryHandler with the maximum retries and reopens set to different values.
+   *
+   * @param maxRetries maximum number of retries
+   * @param maxReopens maximum number of reopens
+   *
+   * @deprecated use CloudStorageRetryHandler(CloudStorageConfiguration) instead.
+   */
+  public CloudStorageRetryHandler(final int maxRetries, final int maxReopens) {
+    this.maxRetries = maxRetries;
+    this.maxReopens = maxReopens;
+    // we're just using the retry parameters from the config, so it's OK to have a default.
+    this.config = CloudStorageConfiguration.DEFAULT;
+  }
 
   /**
    * Create a CloudStorageRetryHandler with the maximum retries and reopens set to the same value.
    *
-   * @param maxRetriesAndReopens value for both maxRetries and maxReopens
+   * @param config - configuration for reopens and retryable codes.
    */
-  public CloudStorageRetryHandler(final int maxRetriesAndReopens) {
```

Comment:
We're still removing this constructor with a single int used for both retries and reopens.

Reply:
We should put this one back and also mark it as deprecated.

Reply:
Oh you're right, thank you for catching that! I put it back, deprecated.